src/main.ts

```diff
@@ -26,17 +26,22 @@ export const run = async () => {
         let inlineScript: string = core.getInput('inlineScript', { required: true });
         let azcliversion: string = core.getInput('azcliversion', { required: false }).trim().toLowerCase();
 
-        if(azcliversion == AZ_CLI_VERSION_DEFAULT_VALUE){
-            const { stdout, stderr } = await cpExec('az version');
-            if (!stderr) {
-                try {
-                    azcliversion = JSON.parse(stdout)["azure-cli"]
-                }
-                catch (er) {
+        if (azcliversion == AZ_CLI_VERSION_DEFAULT_VALUE) {
+            try {
+                const { stdout, stderr } = await cpExec('az version');
+                if (!stderr) {
+                    try {
+                        azcliversion = JSON.parse(stdout)["azure-cli"]
+                    }
+                    catch (er) {
+                        console.log('Failed to fetch az cli version from agent. Reverting back to latest.')
+                        azcliversion = 'latest'
+                    }
+                } else {
                     console.log('Failed to fetch az cli version from agent. Reverting back to latest.')
                     azcliversion = 'latest'

```

Comment:
throw stderr to reduce code duplicacy

Reply:
As this is out internal logic the error thrown should not reflect what we do internally.

Reply:
your thoughts ?

Reply:
if you throw stderr from here the outer catch in line 38 will catch the error